active_discussion/embedded/static-unsynchronized.md

```diff
@@ -0,0 +1,82 @@
+# Unsynchronized access to a shared `static mut` variable
+
+Sometimes interrupt handlers need to share state. This is done using a static
+variable. We claim that *no* synchronization is needed to access the shared
+static variable if the interrupt handlers run at the same priority.
+
+Note that:
+
+- The programs in this document target the [Basic Embedded Abstract Machine
+  (BEAM)][beam]. Please become familiar with the linked specification before you
+  read the rest of this document.
+
+[beam]: https://github.com/rust-lang/unsafe-code-guidelines/pull/111
+
+- In these programs we assume that [rust-lang/rfcs#2585][rfc2585] has been
+  accepted and implemented.
+
+[rfc2585]: https://github.com/rust-lang/rfcs/pull/2585
+
+## Example program
+
+Consider this program
+
+``` rust
+#![no_std]
+
+#[no_mangle]
+unsafe fn main() -> ! {
+    unsafe {
+        asm!("ENABLE_INTERRUPTS" : : : : "volatile");
+    }
+
+    loop {
+        // .. any safe code ..
+    }
+}
+
+static mut X: u128 = 0;
+
+#[no_mangle]
+unsafe fn INTERRUPT0() {
+    let x: &mut u128 = unsafe { &mut X };
+
+    // .. any safe code ..
```

Comment:
Can interrupt0 be interrupted here again by interrupt1 ?

Reply:
Not because:

- they start with the same priority (`1`) when `main` starts executing
- their priority has not been changed in `main` (or elsewhere)
- changing the priority is an `unsafe` operation, which ".. any safe code .." cannot perform.